pkg/types/types.go

```diff
@@ -374,11 +372,13 @@ func mergeMetrics(metrics []Metric, replicaMismatchConfig cfg.RenderReplicaMisma
 
 	// metrics[0] has the highest resolution of metrics
 	metric = metrics[0]
+	valuesForPoint := make([]float64, 0, len(metrics))
+	isMismatchFindConfig := replicaMatchMode != cfg.ReplicaMatchModeNormal
 	for i := range metric.Values {
 		pointExists := !metric.IsAbsent[i]
-		shouldLookForMismatch := replicaMatchMode != cfg.ReplicaMatchModeNormal
+		shouldLookForMismatch := isMismatchFindConfig
 		mismatchObserved := false
-		var valuesForPoint []float64
+		valuesForPoint = valuesForPoint[:0]
```

Comment:
nice catch! this looks like this the true cause. it was allocating small slice objects in a tight loop, which probably trigger higher gc pressures later on?

Reply:
It had impacts but it was not the main one. The bigger overhead was `equalityFunc` function calls. Making exact equality check inline, had a bigger impact.

Reply:
hmm, really? that's a bit surprising.

Reply:
nice!

probably made sense to drop the `equalityFunc` in the future. really no need to support approximate comparison. whisper files has deterministic aggregation/rollup behaviour.